app/scripts/controllers/network/network-controller.test.ts

```diff
@@ -1,64 +1,195 @@
+ * Note that this type does not specify all of the properties present within the
+ * block header; within these tests, we are only interested in `number` and
+ * `baseFeePerGas`.
  */
+type Block = {
+  number: string;
+  baseFeePerGas?: string;
+};
 
 /**
- * @typedef {{request: MockJsonResponseBody, response: { httpStatus?: number } & MockJsonResponseBody, error?: unknown, delay?: number; times?: number, beforeCompleting: () => void | Promise<void>}} RpcMock
- *
- * Arguments to `mockRpcCall` which allow for specifying a canned response for a
- * particular RPC request.
+ * A partial form of a prototypical JSON-RPC request body.
  */
+type MockJsonRpcRequestBody = {
+  id?: number;
+  jsonrpc?: string;
+  method: string;
+  params?: unknown[];
+};
 
 /**
- * @typedef {{id?: number; jsonrpc?: string, method: string, params?: unknown[]}} MockJsonRpcRequestBody
- *
- * A partial form of a prototypical JSON-RPC request body.
+ * A composite form of a prototypical JSON-RPC response body.
  */
+type MockJsonRpcResponseBody = {
+  id?: number | string;
+  jsonrpc?: '2.0';
+  result?: unknown;
+  error?: string | null;
+};
 
 /**
- * @typedef {{id?: number; jsonrpc?: string; result?: string; error?: string}} MockJsonResponseBody
- *
- * A partial form of a prototypical JSON-RPC response body.
+ * Arguments to `mockRpcCall` which specify the behavior of a mocked RPC
+ * request.
+ */
+type RpcCallMockSpec = {
+  request: MockJsonRpcRequestBody;
+  response: MockJsonRpcResponseBody & { httpStatus?: number };
+  error?: unknown;
+  delay?: number;
+  times?: number;
+  beforeCompleting?: () => void | Promise<void>;
+};
+
+/**
+ * A partial form of `RpcCallMockSpec`, which is preferred in
+ * `mockEssentialRpcCalls` for brevity.
  */
+type PartialRpcCallMockSpec = {
+  request?: Partial<MockJsonRpcRequestBody>;
+  response?: Partial<MockJsonRpcResponseBody & { httpStatus?: number }>;
+  error?: unknown;
+  delay?: number;
+  times?: number;
+  beforeCompleting?: () => void | Promise<void>;
+};
+
+/**
+ * An RPC method that `mockEssentialRpcCalls` recognizes.
+ */
+enum KnownMockableRpcMethod {
+  EthBlockNumber = 'eth_blockNumber',
+  EthGetBlockByNumber = 'eth_getBlockByNumber',
+  NetVersion = 'net_version',
+}
+
+/**
+ * The callback that `withController` takes.
+ */
+type WithControllerCallback<NetworkCommunications, ReturnValue> = (args: {
+  controller: NetworkController;
+  network: NetworkCommunications;
+}) => Promise<ReturnValue> | ReturnValue;
+
+/**
+ * The options that `withController` takes.
+ */
+type WithControllerOptions = Partial<NetworkControllerOptions>;
+
+/**
+ * A variant of the options that the NetworkController constructor takes, where
+ * the provider state has been preconfigured with an Infura network. This is
+ * extracted so that we can give `withController` a better signature.
+ */
+type NetworkControllerOptionsWithInfuraProviderConfig = Partial<
+  Omit<WithControllerOptions, 'state'>
```

Comment:
I thought so too. I was having trouble with this when I was updating `withController` and must have thought this would help TypeScript. You're right though that it's unnecessary. Updated in [4dd0879](https://github.com/MetaMask/metamask-extension/pull/18476/commits/4dd0879fcc718415eaa2de8d333ca9b2871df2ce).